### PR TITLE
Section dialog fixes

### PR DESCRIPTION
**Pull request by larrypo**
Prevents showing section code in students of sections for Clever and Google Classroom managed sections.

### CODE REVIEW

File: apps/src/templates/studioHomepages/SectionsTable.jsx [hunk 1]
```diff
@@ -206,7 +207,9 @@ const SectionsTable = React.createClass({
                 </td>
               )}
               <td style={{...styles.col, ...(isRtl? styles.sectionCodeColRtl: styles.sectionCodeCol)}}>
-                {section.code}
+                <nobr>{section.login_type == SectionLoginType.clever ? i18n.loginTypeClever() :
+                    section.login_type == SectionLoginType.google_classroom ? i18n.loginTypeGoogleClassroom() :
+                        section.code}</nobr>
```
Comment (islemaster) [inline apps/src/templates/studioHomepages/SectionsTable.jsx]: I'd prefer we used the `white-space: nowrap` CSS property here to the `<nobr>` tag, [which is apparently nonstandard](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nobr).

**Comment by larrypo** [comment]
Done.

**Comment by islemaster** [comment]
Test failure is that our linter requires strict equality `===` [here and on the line below](https://github.com/code-dot-org/code-dot-org/pull/18220/files#diff-27925f9eabdd9bbab879b3897b6006cdR212).  I'll start another build as soon as you can push a change.

**Comment by larrypo** [comment]
Done.

**Comment by islemaster** [comment]
I'm seeing a unit test failure that seems to result from the change to the `sections` shape in this PR:
```
FAILED TESTS:
  unit tests
    ✖ "after each" hook for "shows a StudentSections component"
      PhantomJS 2.1.1 (Linux 0.0.0)
    Error: Call to console.error from "shows ProjectWidgetWithData component": Warning: Failed prop type: The prop `sections[0].login_type` is marked as required in `StudentHomepage`, but its value is `undefined`.
        in StudentHomepage
```

**Comment by islemaster** [comment]
Latest build failed on a simple lint error.  I've [pushed a fix](https://github.com/code-dot-org/code-dot-org/commit/4dbc969c8d7809fd33d73158b79da4ccafb0a043) to the branch I'm running tests from.  If you merge `code-dot-org/larrypo_section_dialog_fixes` into your branch from your own client then push your branch, this PR should automatically attach to the new test run at that commit.

**Comment by islemaster** [comment]
[Tests passed](https://circleci.com/gh/code-dot-org/code-dot-org/51370) after the lint fix - I've merged this changeset + that fix into staging, closing this PR.  Thanks Larry!

**Pull request merged**